python/samples/demos/guided_conversations/interactive_guided_conversation.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+"""Run this interactive guided conversation script to test out the teaching scenario!
+The teaching artifact, rules, conversation flow, context, and resource constraint can all be modified to
+fit your needs & try out new scenarios!
+"""
+
+import asyncio
+
+from pydantic import BaseModel, Field
+from semantic_kernel import Kernel
+
+from guided_conversation.guided_conversation_agent import GCInput, GuidedConversation
+from guided_conversation.utils.resources import ResourceConstraint, ResourceConstraintMode, ResourceConstraintUnit
+from guided_conversation.utils.services import add_service
+
+
+# Artifact - The artifact is like a form that the agent must complete throughout the conversation.
+# It can also be thought of as a working memory for the agent.
+# We allow any valid Pydantic BaseModel class to be used.
+class MyArtifact(BaseModel):
```

Comment:
Same question here: can we leverage SK's `KernelBaseModel` here?

Reply:
I don't think we need that here, KernelBaseModel is a opinitated BaseModel used within SK, not necessarily something that people should build on even for a sample, since we might change it and that should not break client code!